### ISSUE TITLE
[Bug][3.5.0] Pinia store not persisting to client-side

### ISSUE THREAD

**Issue by silverbackdan** [nuxt/nuxt#20941]
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.16.0`
- Nuxt Version:     `3.5.0`
- Nitro Version:    `2.4.1`
- Package Manager:  `pnpm@8.5.1`
- Builder:          `vite`
- User Config:      `modules`, `cwa`, `extends`, `srcDir`, `typescript`
- Runtime Modules:  `nuxt-vitest@0.7.0`
- Build Modules:    `-`

### Reproduction

I'm not sure how to make a minimal reproduction right now, but it is apparent in my module. Using 3.4.* I have my store data populated on server-side and shows on client side. When upgrading to 3.5.0 the SSR shows the data and then disappears on client-side

### Describe the bug

As described above. I also have some video reproductions.
> v 3.5.0

https://github.com/nuxt/nuxt/assets/25174262/e6008423-3cf6-453f-be3d-f41ee32a3155

> v 3.4.x

https://github.com/nuxt/nuxt/assets/25174262/f87ddae5-dc85-4b3c-9a81-c31ec1bd2441

### Additional context

I am not implementing stores in a standard way I suppose. But still there is a BC break.

My storage classes are initialised here:
https://github.com/components-web-app/cwa-nuxt-module/tree/nuxt3/src/runtime/storage

And this is initialised from a module which includes a plugin that creates a class instance of CWA which will create a new storage class in the constructor.

**Comment by danielroe** [comment]
Would you try defining a custom reviver + reducer with `definePayloadReducer` and `definePayloadReviver`? See example at https://nuxt.com/docs/api/composables/use-nuxt-app#payload.

You probably can work around this as well by disabling `experimental.renderJsonPayloads`.

**Comment by Krutie** [comment]
Hey @silverbackdan 

I see that you are using Nuxt Dev Tool. Please try disabling Nuxt Dev Tool and see if this issue gets resolved. 

## Issue 

I had similar issue where the Pinia store shows data when printing `window.__NUXT__.pinia` in console, and even in `nuxtApp.payload.data` - but then nothing is available when accessing Pinia store (`productStore.data`) on the client-side.  Data flashes on the screen for a split second, then disappears and throw `Hydration completed but contains mismatches` error in console. 

If disabling doesn't work, remove Nuxt Dev Tool altogether.  Doing this somehow, stopped resetting Pinia store state. HTH.

Versions:
- "nuxt": "^3.5.0"
- "@pinia/nuxt": "^0.4.11",
- "pinia": "^2.1.1"

**Comment by danielroe** [comment]
Does someone have a reproduction for where devtools are causing this issue? If we can raise it upstream on https://github.com/nuxt/devtools hopefully we can resolve. (cc: @antfu)

**Comment by Krutie** [comment]
> Does someone have a reproduction for where devtools are causing this issue? If we can raise it upstream on https://github.com/nuxt/devtools hopefully we can resolve. (cc: @antfu)

Hey @danielroe / @antfu 
Stackblitz does not replicate the issue being discussed here. :(

I've created this repo with a branch - with steps in the README file - to reproduce the bug.
https://github.com/Krutie/nuxt-3-lessons/tree/issue-20941

**Comment by silverbackdan** [comment]
Mine appears to be a little more complex. I removed dev tools and then got the error 
```
"getActivePinia()" was called but there was no active Pinia
```

This had started happening in tests since the latest update and then I updated my plugin to set active pinia in test mode - however I see after removing dev tools, that this could be the root cause. Some update has resulted in me trying to call a pinia store, from a constructor in a ref, after my stores are setup, but results in this error.

So perhaps the dev tools is suppressing this error somehow in my use case, or setting an active pinia instance automatically when my module/plugin seems to have an issue..

Perhaps something in a Nuxt Pinia module update is simply resulting in the active pinia store not always being set and then the dev tools is setting again... or something...

**Comment by silverbackdan** [comment]
Mine appears to be a little more complex. I removed dev tools and then got the error

```
"getActivePinia()" was called but there was no active Pinia. Did you forget to install pinia? const pinia = createPinia() app.use(pinia) This will fail in production.
```

This had started happening in tests since the latest update and then I updated my plugin to set active pinia in test mode - however I see after removing dev tools, that this could be the root cause. Some update has resulted in me trying to call a pinia store, from a constructor in a ref, after my stores are setup, but results in this error.

So perhaps the dev tools is suppressing this error somehow in my use case, or setting an active pinia instance automatically when my module/plugin seems to have an issue..

Perhaps something in a Nuxt Pinia module update is simply resulting in the active pinia store not always being set (or in my case, not set in time) and then the dev tools is setting again...
